src/sentry/api/endpoints/organization_details.py

```diff
@@ -248,6 +297,55 @@ def validate(self, attrs):
                 )
         return attrs
 
+    def save_trusted_relays(self, incoming, changed_data, organization):
+        timestamp_now = (
+            datetime.utcnow().isoformat() + "Z"
+        )  # make it utc relative with set timezone
+        option_key = "sentry:trusted-relays"
+        try:
+            # get what we already have
+            existing = OrganizationOption.objects.get(organization=organization, key=option_key)
+            if existing is not None:
+                key_dict = {val.get("public_key"): val for val in existing.value}
+            else:
+                key_dict = {}
+        except OrganizationOption.DoesNotExist:
+            key_dict = {}  # we don't have anything set
+            existing = None
+
+        modified = False
+        for option in incoming:
+            public_key = option.get("public_key")
+            existing_info = key_dict.get(public_key, {})
+
+            option["created"] = existing_info.get("created", timestamp_now)
+            option["last_modified"] = existing_info.get("last_modified")
+
+            # check if we modified the current public_key info and update last_modified if we did
+            if (
+                existing_info.get("public_key") is None
+                or existing_info.get("name") != option.get("name")
+                or existing_info.get("description") != option.get("description")
+            ):
+                option["last_modified"] = timestamp_now
+                modified = True
+
+        if modified:
```

Comment:
I believe that `modified` is never set if we are only removing a previously existing trusted Relay.

Reply:
you are right, modified will not be set if you remove *all* keys, now it is fixed.